[BUG] org.opensearch.search.nested.SimpleNestedExplainIT.testExplainMultipleDocs is flaky

### Describe the bug

The test case `org.opensearch.search.nested.SimpleNestedExplainIT.testExplainMultipleDocs` is flaky:

```
org.opensearch.search.nested.SimpleNestedExplainIT.testExplainMultipleDocs


java.lang.AssertionError: expected:<0.36464313 = Score based on 2 child docs in range from 0 to 1, using score mode Total
  0.18232156 = weight(nested1.n_field1:n_value1 in 0) [PerFieldSimilarity], result of:
    0.18232156 = score(freq=1.0), computed as boost * idf * tf from:
      2.2 = boost
      0.18232156 = idf, computed as log(1 + (N - n + 0.5) / (n + 0.5)) from:
        2 = n, number of documents containing term
        2 = N, total number of documents with field
      0.45454544 = tf, computed as freq / (freq + k1 * (1 - b + b * dl / avgdl)) from:
        1.0 = freq, occurrences of term within document
        1.2 = k1, term saturation parameter
        0.75 = b, length normalization parameter
        1.0 = dl, length of field
        1.0 = avgdl, average length of field
> but was:<0.36464313 = Score based on 2 child docs in range from 14 to 15, using score mode Total
  0.18232156 = weight(nested1.n_field1:n_value1 in 14) [PerFieldSimilarity], result of:
    0.18232156 = score(freq=1.0), computed as boost * idf * tf from:
      2.2 = boost
      0.18232156 = idf, computed as log(1 + (N - n + 0.5) / (n + 0.5)) from:
        2 = n, number of documents containing term
        2 = N, total number of documents with field
      0.45454544 = tf, computed as freq / (freq + k1 * (1 - b + b * dl / avgdl)) from:
        1.0 = freq, occurrences of term within document
        1.2 = k1, term saturation parameter
        0.75 = b, length normalization parameter
        1.0 = dl, length of field
        1.0 = avgdl, average length of field
>
```

### Related component

Search

### To Reproduce

```
./gradlew ':server:internalClusterTest' --tests "org.opensearch.search.nested.SimpleNestedExplainIT.testExplainMultipleDocs" -Dtests.seed=52C9CA95BCB30A28 
```

### Expected behavior

The test must always pass

### Additional Details

**Plugins**
Standard

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Host/Environment (please complete the following information):**
 - CI

**Additional context**
- https://build.ci.opensearch.org/job/gradle-check/33757/testReport/junit/org.opensearch.search.nested/SimpleNestedExplainIT/testExplainMultipleDocs/

--- a/server/src/internalClusterTest/java/org/opensearch/search/nested/SimpleNestedExplainIT.java
+++ b/server/src/internalClusterTest/java/org/opensearch/search/nested/SimpleNestedExplainIT.java
@@ -30,6 +30,11 @@ import static org.hamcrest.Matchers.equalTo;
  */
 public class SimpleNestedExplainIT extends OpenSearchIntegTestCase {
 
+    @Override
+    protected int numberOfShards() {
+        return 1;
+    }
+
     /*
      * Tests the explain output for multiple docs. Concurrent search with multiple slices is tested
      * here as call to indexRandomForMultipleSlices is made and compared with explain output for
@@ -70,7 +75,23 @@ public class SimpleNestedExplainIT extends OpenSearchIntegTestCase {
             .setRefreshPolicy(IMMEDIATE)
             .get();
 
-        indexRandomForMultipleSlices("test");
+        client().prepareIndex("test")
+            .setId("2")
+            .setSource(
+                jsonBuilder().startObject()
+                    .field("field1", "value2")
+                    .startArray("nested1")
+                    .startObject()
+                    .field("n_field1", "n_value2")
+                    .endObject()
+                    .startObject()
+                    .field("n_field1", "n_value2")
+                    .endObject()
+                    .endArray()
+                    .endObject()
+            )
+            .setRefreshPolicy(IMMEDIATE)
+            .get();
 
         // Turn off the concurrent search setting to test search with non-concurrent search
         client().admin()

